pkgs/applications/video/anime-downloader/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib, python3Packages, aria2, mpv, nodejs, fetchFromGitHub }:
+with python3Packages;
+buildPythonApplication rec {
+  pname = "anime-downloader";
+  version = "5.0.14";
+  src = fetchFromGitHub ({
+    owner = "anime-dl";
+    repo = "anime-downloader";
+    rev = "a509e4a8054665180bb2f172b3c0883a5a7da0ce";
+    sha256 = "sha256-Uk2mtsSrb8fCD9JCFzvLBzMEB7ViVDrKPSOKy9ALJ6o=";
+  });
+
+  propagatedBuildInputs = [
+    aria2
+    beautifulsoup4
+    cfscrape
+    click
+    coloredlogs
+    fuzzywuzzy
+    jsbeautifier
+    mpv
+    nodejs
+    pySmartDL
+    pycryptodome
+    pyqt5
+    requests
+    requests-cache
+    selenium
+    tabulate
+  ];
+
+  doCheck = false;
```

Comment:
could you mention why ?

Reply:
I must fix some things. :disappointed:  

By doing checks i got this:

```sh
error: builder for '/nix/store/28cw3finqanlb9gjl55gnl3vvyqnsjd0-anime-downloader-5.0.14.drv' failed with exit code 1;
       last 10 log lines:
       >     mkdir(name, mode)
       > PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
       >
       >
       > ----------------------------------------------------------------------
       > Ran 1 test in 0.000s
       >
       > FAILED (errors=1)
       > Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
       > error: Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
       For full logs, run 'nix log /nix/store/28cw3finqanlb9gjl55gnl3vvyqnsjd0-anime-downloader-5.0.14.drv'.
```

I could get around it by adding this line:

```nix
{
  preCheck = ''
    export HOME=$PWD/home
  '';
}
```

But then I got this error (which is probably why `anime gui` wasn't working):

```sh
error: builder for '/nix/store/kmc61n5gyhjg8vc3xwvw2wbkwg6s3fsh-anime-downloader-5.0.14.drv' failed with exit code 134;
       last 10 log lines:
       > writing requirements to anime_downloader.egg-info/requires.txt
       > writing top-level names to anime_downloader.egg-info/top_level.txt
       > reading manifest file 'anime_downloader.egg-info/SOURCES.txt'
       > adding license file 'LICENSE'
       > writing manifest file 'anime_downloader.egg-info/SOURCES.txt'
       > running build_ext
       > qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
       > This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
       >
       > /nix/store/fsskcsd0pbnlsw8j63w1ninbdcg82lb5-setuptools-check-hook/nix-support/setup-hook: line 4:   388 Aborted                 (core dumped) /nix/store/553d7c4xcwp9j1a1gb9cb1s9ry3x1pi9-python3-3.9.13/bin/python3.9 nix_run_setup test
       For full logs, run 'nix log /nix/store/kmc61n5gyhjg8vc3xwvw2wbkwg6s3fsh-anime-downloader-5.0.14.drv'.
```

So for now I need a way to "wrap" QT paths inside the package I guess...

Reply:
Seems like a common problem: https://github.com/NixOS/nixpkgs/blob/634141959076a8ab69ca2cca0f266852256d79ee/pkgs/applications/misc/openlp/lib.nix#L20-L23

Reply:
this seems non trivial to fix so you could just reference the error to justify the disabling. Sometimes it's also possible to skip the failing tests (see other python packages on how to do this, it depends on the testing framework).